File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
N = int(input())
bag = 0
while N >= 0 :
    if N % 5 == 0 :  # 5의 배수이면
        bag += (N // 5)  
        print(bag)
        break
    N -= 3  # 5의 배수가 될 때까지 
    bag += 1  
else :
    print(-1)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int bag = 0;
        while(true) {
            if(N%5 == 0) {
                bag += N/5;
                System.out.println(bag);
                break;
            }
            else {
                N -= 3;
                bag++;
            }
            if(N < 0) {
                System.out.println("-1");
                break;
            }
        }
    }
}